impl/src/test/groovy/io/jsonwebtoken/impl/security/KeysBridgeTest.groovy

```diff
@@ -56,4 +60,46 @@ class KeysBridgeTest {
     void testToStringPassword() {
         testFormattedOutput(new PasswordSpec("foo".toCharArray()))
     }
+
+    @Test
+    void testIsGenericSecret() {
+        def genericSecret = new SecretKey() {
+            @Override
+            String getAlgorithm() {
+                return "GenericSecret" ;
```

Comment:
We'd probably want another test/tests with a space between 'Generic' and 'Secret' if we go with the .startsWith/.endsWith approach.